### DISCUSSION TITLE
First Step in XKNX - IP Connextion

### DISCUSSION THREAD

**Discussion by kepir** [XKNX/xknx#1004] [Q&A]
Hi to all,
I just want to Connect to my Siemens N146/02 Gateway. 
I have a working Python environment and the KNX Bus is big ( around 300 Devices )and also working well.
The ip Adress is for example 172.16.0.195 and its port is 3671.

I need some tips to connect and listen to the Bus Monitor or turn licht on or off.

Thank you

**Comment by farmio** [comment]
Hi 👋! 
Have you had a look at our documentation site https://xknx.io and the example scripts? 
https://github.com/XKNX/xknx/tree/main/examples

**Reply by kepir** [reply]
Yes i did .
But not find the place where i can set my gateway ip .

**Reply by farmio** [reply]
See here: https://github.com/XKNX/xknx/blob/main/examples/example_manual_connection.py
It shouldn't be necessary though as it should be automatically discovered.

**Comment by kepir** [comment]
I know,  but I have remote NAT connection. I need to set the up manually.

Kepir

**Reply by farmio** [reply]
Use `route_back=True` then on the connection config.

**Comment by kepir** [comment]
Where is the connection config file ?

**Reply by farmio** [reply]
Here: https://github.com/XKNX/xknx/blob/main/xknx/io/connection.py

Have a look at this example: https://github.com/XKNX/xknx/blob/main/examples/example_manual_connection.py

**Comment by TheSundayDev** [comment]
hi all, 
first of all thanls for this amazing library!
I'm using TUNNELING connection with this config:
connection_config = ConnectionConfig(
        connection_type=ConnectionType.TUNNELING,
        gateway_ip=gatewayIp,
        gateway_port=gatewayPort,
        route_back=True)
Often, not always but very often, I have this error:
CommunicationError: Tunnel connection could not be established Traceback
What can i do?
thanks

**Reply by farmio** [reply]
Hmm... that depends on the actual error that's printed there. Look for a "Status code" or something like that.

**Reply by TheSundayDev** [reply]
Hi, thankyou for your quick response.
I see just this:
[ERROR] CommunicationError: Tunnel connection could not be established
Traceback (most recent call last):
  File "/var/lang/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
  File "/var/lang/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
  File "/var/lang/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
  File "/var/task/lambda_function.py", line 88, in send
    async with XKNX(connection_config=connection_config) as xknx:
  File "/opt/python/xknx/xknx.py", line 103, in __aenter__
    await self.start()
  File "/opt/python/xknx/xknx.py", line 125, in start
    await self.knxip_interface.start()
  File "/opt/python/xknx/io/knxip_interface.py", line 72, in start
    await self._start()
  File "/opt/python/xknx/io/knxip_interface.py", line 100, in _start
    await self._start_tunnelling_udp(
  File "/opt/python/xknx/io/knxip_interface.py", line 381, in _start_tunnelling_udp
    await self._interface.connect()
  File "/opt/python/xknx/io/tunnel.py", line 123, in connect
    raise CommunicationError(

Is there any timeout setting for the connection?
Because this happens when I run it from the cloud, from local is working fine.

**Reply by farmio** [reply]
This traceback should be longer. Somewhere at the end you should see what caused the CommunicationError.
Otherwise, you may get some insights using Wireshark (`kip` display filter).
> Is there any timeout setting for the connection?

No. We use constant timeout values according to the specifications.

> Because this happens when I run it from the cloud, from local is working fine.

You probably want to use `route_back` then as there is most likely NAT involved. 
Also, just in case: never forward a port of a KNX IP interface to the internet - use some VPN or something.
